client/transport.c

```diff
@@ -42,19 +42,64 @@ struct xdag_send_data {
 	struct send_parameters send_parameters;
 };
 
-struct xdag_task_info {
-	uint8_t task_type;
-	uint32_t block_req_counter;
+enum task_info_action {
+	TASK_INFO_START,
+	TASK_INFO_END
 };
 
 enum task_type {
-	TASK_REQBLOCKS		= 0x01,
-	TASK_REQBLOCKS_THREAD	= 0x02,
-	TASK_REQSUM		= 0x04
+	TASK_REQBLOCKS			= 0x01,
```

Comment:
0x01 (hex) = 1 (decimal), so TASK_REQBLOCKS and TASK_REQBLOCKS_THREAD_INDEX have the same indexes. Are you sure you didn't make error here? 
Maybe you use hex values for all enum values?